examples_test.go

```diff
@@ -0,0 +1,288 @@
+//       that wasn't created via eris.NewGlobal (e.g. the root error stack will be
+//       wrong and might cause issues when trying to insert wrap frames into it)
+
+// Demonstrates JSON formatting of wrapped errors that originate from
+// external (non-eris) error types.
+func ExampleUnpackedError_ToJSON_external() {
+	// example func that returns an IO error
+	readFile := func(fname string) error {
+		return io.ErrUnexpectedEOF
+	}
+
+	// unpack and print the error
+	err := readFile("example.json")
+	uerr := eris.Unpack(err)
+	format := eris.NewDefaultFormat(false) // false: omit stack trace
+	u, _ := json.Marshal(uerr.ToJSON(format))
+	fmt.Println(string(u))
+	// Output:
+	// {"external":"unexpected EOF"}
+}
+
+// Demonstrates JSON formatting of wrapped errors that originate from
+// global root errors (created via eris.NewGlobal).
+func ExampleUnpackedError_ToJSON_global() {
+	// declare a "global" error type
+	ErrUnexpectedEOF := eris.NewGlobal("unexpected EOF")
+
+	// example func that wraps a global error value
+	readFile := func(fname string) error {
+		return eris.Wrapf(ErrUnexpectedEOF, "error reading file '%v'", fname)
+	}
+
+	// example func that catches and returns an error without modification
+	parseFile := func(fname string) error {
+		// read the file
+		err := readFile(fname)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	// unpack and print the error via uerr.ToJSON(...)
+	err := parseFile("example.json")
+	uerr := eris.Unpack(err)
+	format := eris.NewDefaultFormat(true) // true: include stack trace
+	u, _ := json.MarshalIndent(uerr.ToJSON(format), "", "\t")
+	fmt.Printf("%v\n", string(u))
+
+	// example output:
+	// {
+	//   "root": {
+	//     "message": "unexpected EOF",
+	//     "stack": [
+	//       "eris_test.ExampleUnpackedError_ToJSON_global.func1: /Users/morningvera/go/src/github.com/rotisserie/eris/examples_test.go: 41",
+	//       "eris_test.ExampleUnpackedError_ToJSON_global.func2: /Users/morningvera/go/src/github.com/rotisserie/eris/examples_test.go: 47",
+	//       "eris_test.ExampleUnpackedError_ToJSON_global: /Users/morningvera/go/src/github.com/rotisserie/eris/examples_test.go: 55",
+	//       "eris_test.TestExampleUnpackedError_ToJSON_global: /Users/morningvera/go/src/github.com/rotisserie/eris/examples_test.go: 87"
+	//     ]
+	//   },
+	//   "wrap": [
+	//     {
+	//       "message": "error reading file 'example.json'",
+	//       "stack": "eris_test.ExampleUnpackedError_ToJSON_global.func1: /Users/morningvera/go/src/github.com/rotisserie/eris/examples_test.go: 41"
+	//     }
+	//   ]
+	// }
+}
+
+// Hack to run examples that don't have a predictable output (i.e. all
+// examples that involve printing stack traces).
+func TestExampleUnpackedError_ToJSON_global(t *testing.T) {
+	if !testing.Verbose() {
+		return
+	}
+	ExampleUnpackedError_ToJSON_global()
+}
+
+// Demonstrates JSON formatting of wrapped errors that originate from
```

Comment:
add go playground links

Reply:
it seems like godoc doesn't handle links in example comments. i think we should defer this to another time if we still want to figure out how to make it work